util/types/time.go

```diff
@@ -326,8 +329,15 @@ func (t Time) RoundFrac(fsp int) (Time, error) {
 	}
 
 	var nt TimeInternal
-	// TODO: Consider time_zone variable.
-	if t1, err := t.Time.GoTime(gotime.Local); err == nil {
+	var loc *gotime.Location
+	if t.Type == mysql.TypeTimestamp {
+		loc = t.TimeZone
+	} else {
+		// TODO: Consider time_zone variable.
+		loc = gotime.Local
```

Comment:
Timestamp datum has a timezone, while Datetime/Date doesn't.
So convert Timestamp to Go's could use the timezone, but in other cases I have to provide a timezone.